core/trie/proof.go

```diff
@@ -0,0 +1,80 @@
+package trie
+
+import (
+	"errors"
+
+	"github.com/NethermindEth/juno/core/crypto"
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+type Membership int
+
+const (
+	Member Membership = iota
+	NonMember
+)
+
+type ProofNode struct {
+	LeftHash  *felt.Felt
+	RightHash *felt.Felt
+}
+
+// https://github.com/eqlabs/pathfinder/blob/main/crates/merkle-tree/src/tree.rs#L514
+func GetProof(leaf *felt.Felt, tri *Trie) ([]ProofNode, error) {
+	leafKey := tri.feltToKey(leaf)
+	nodesToLeaf, err := tri.nodesFromRoot(&leafKey)
+	if err != nil {
+		return nil, err
+	}
+	proofNodes := make([]ProofNode, len(nodesToLeaf)-1)
+
+	getHash := func(tri *Trie, key *Key) (*felt.Felt, error) {
```

Comment:
`tri` is always the same: wdyt about removing this parameter and rely on closure (i.e. capturing outer scope with `tri` parameter from `GetProof(...)` ?